src/OrchardCore/OrchardCore.Logging.NLog/WebHostBuilderExtensions.cs

```diff
@@ -18,14 +18,19 @@ public static IWebHostBuilder UseNLogWeb(this IWebHostBuilder builder)
             .ConfigureAppConfiguration((context, _) =>
             {
                 var environment = context.HostingEnvironment;
-                var appData = System.Environment.GetEnvironmentVariable(ShellOptionConstants.OrchardAppData);
-                var configDir = string.IsNullOrWhiteSpace(appData) ? $"{environment.ContentRootPath}/{ShellOptionConstants.DefaultAppDataPath}" : appData;
-                LogManager.Configuration.Variables["configDir"] = environment.ContentRootPath;
+                var appData = System.Environment.GetEnvironmentVariable(ShellOptionConstants.OrchardAppData);                
+                var configDir = string.IsNullOrWhiteSpace(appData) ? Path.Combine(environment.ContentRootPath, ShellOptionConstants.DefaultAppDataPath) : appData;
+                if (LogManager.Configuration == null)
+                {
+                    LogManager.Configuration = new LoggingConfiguration();
```

Comment:
Maybe just perform null-check, instead of assigning dummy `LoggingConfiguration`:
```c#
var loggerConfiguration = LogManager.Configuration;
if (loggerConfiguration is not null)
    loggerConfiguration.Variables["configDir"] = configDir;
```